SciPy imposes strict and unnecessary version restrictions on its dependencies.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -35,7 +35,7 @@
     #      and disabling build isolation.
     #   3. The <2.3 upper bound is for matching the numpy deprecation policy,
     #      it should not be loosened.
-    "numpy>=2.0.0rc1,<2.3",
+    "numpy",
 ]
 
 [project]
